app/models/problem_check.rb

```diff
@@ -106,9 +133,40 @@ def call
     raise NotImplementedError
   end
 
+  def run
+    problems = call
+
+    yield(problems) if block_given?
+
+    next_run_at = perform_every&.from_now
+
+    if problems.empty?
+      targets.each { |t| tracker(t).no_problem!(next_run_at:) }
+    else
+      problems
+        .uniq(&:target)
+        .each do |problem|
+          tracker(problem.target).problem!(
+            next_run_at:,
+            details: translation_data.merge(problem.details).merge(base_path: Discourse.base_path),
+          )
+        end
+    end
+
+    problems
+  end
+
   private
 
-  def problem(override_key = nil, override_data = {})
+  def tracker(target = nil)
+    ProblemCheckTracker[identifier, target]
+  end
+
+  def targets
+    [nil]
```

Comment:
Why `[nil]` instead of `[]` here?

Reply:
It simplifies this line for trackers that don't have targets: https://github.com/discourse/discourse/pull/26192/files#diff-de03beaf4c0ac50d9abc3dc86929a54d7cf216436d621e451f040c2dd0db19daR144